### PR TITLE
csi: nomad volume detach command

### PR DESCRIPTION
**Pull request by tgross**
For https://github.com/hashicorp/nomad/issues/8252

The soundness guarantees of the CSI specification leave a little to be desired
in our ability to provide a 100% reliable automated solution for managing
volumes. This changeset provides a new command to bridge this gap by providing
the operator the ability to intervene.

The command doesn't take an allocation ID so that the operator doesn't have to
keep track of alloc IDs that may have been GC'd. Handle this case in the
unpublish RPC by sending the client RPC for all the terminal/nil allocs on the
selected node.

### CODE REVIEW

File: website/pages/api-docs/volumes.mdx [hunk 1]
```diff
@@ -323,6 +323,12 @@ The table below shows this endpoint's support for
   running allocations. This does not detach the volume from client nodes.
   This is specified as a query string parameter.
 
+- `detach` `(bool: false)`- Detach the volume from the node given by the `node`
```
Comment (tgross) [inline website/pages/api-docs/volumes.mdx:326]: I'm really not wild about this approach, but volume HTTP API is fairly "restful" so I'm not really sure what would be a better way to go here. Maybe @DingoEatingFuzz has an opinion here?

Comment (langmartin) [reply website/pages/api-docs/volumes.mdx:326]: `/v1/volume/csi/:volume_id/detach` doesn't have the collision problem

Comment (tgross) [reply website/pages/api-docs/volumes.mdx:326]: My imagination was failing me there.  🤦 

Do we like that better? I think it feels less gross.

Comment (langmartin) [reply website/pages/api-docs/volumes.mdx:326]: I like it a little better, but I don't know that it matters a lot.

The only technical consideration I can think of is I've seen proxies discard a DELETE body, but a) probably no proxy and b) query params may be fine. I've found some strong opinions strongly held that it's bad, but not much evidence that it's actually a problem 😁

Comment (tgross) [reply website/pages/api-docs/volumes.mdx:326]: Per our internal discussion we're going to change this to be `/v1/volume/csi/:volume_id/detach`

Comment (tgross) [reply website/pages/api-docs/volumes.mdx]: Done in 2e3ad64

File: website/pages/docs/commands/volume/detach.mdx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+layout: docs
+page_title: 'Commands: volume detach'
+sidebar_title: detach
+description: |
+  Detach volumes with CSI plugins.
+---
+
+# Command: volume detach
+
+The `volume detach` command detachs external storage volumes with Nomad's
```
Comment (langmartin) [inline website/pages/docs/commands/volume/detach.mdx:11]: is it "detatches"?

File: website/pages/docs/commands/volume/detach.mdx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+## Usage
+
+```plaintext
+nomad volume detach [options] [volume] [node]
+```
+
+The `volume detach` command requires two arguments, specifying the ID of the
+volume to be detached and the node to detach it from. Detaching will fail if
+fail if the volume is still in use by an allocation.
```
Comment (langmartin) [inline website/pages/docs/commands/volume/detach.mdx:23]: dup "fail if"

File: website/pages/docs/commands/volume/detach.mdx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+sidebar_title: detach
+description: |
+  Detach volumes with CSI plugins.
+---
+
+# Command: volume detach
+
+The `volume detach` command detachs external storage volumes with Nomad's
+[Container Storage Interface (CSI)][csi] support. The volume must exist on the
+remote storage provider before it can be detached and used by a task.
```
Comment (langmartin) [inline website/pages/docs/commands/volume/detach.mdx:13]: is this orphaned copy from the register page? It seems technically true but a little confusing here.

Comment (tgross) [reply website/pages/docs/commands/volume/detach.mdx:13]: Fair enough.

**Pull request merged**